src/bun.js/bindings/ZigGlobalObject.cpp

```diff
@@ -4675,6 +4675,21 @@ bool GlobalObject::hasNapiFinalizers() const
     return false;
 }
 
+extern "C" void Zig__GlobalObject__destructOnExit(Zig::GlobalObject* globalObject)
+{
+    if (JSC::getVM(globalObject).entryScope) {
+        // Exiting while running JavaScript code (e.g. `process.exit()`), so we can't destroy it
+        // just now. Perhaps later in this case we can defer destruction to run later.
+        return;
+    }
+    auto& vm = JSC::getVM(globalObject);
+    gcUnprotect(globalObject);
+    globalObject = nullptr;
+    vm.heap.collectNow(JSC::Sync, JSC::CollectionScope::Full);
```

Comment:
Maybe we should do the other things WebCore does when doing a full collection scope? Freeing the threadlocal WTF fast malloc memory?